### PR TITLE
Atomic: update documentation and add institution

### PR DESCRIPTION
**Pull request by gadmm**
This addresses multicore review thread https://github.com/ocaml/ocaml/pull/10831#pullrequestreview-846594578.

From the commit log

- Update outdated documentation.
- Add copyright for Stephen's Institution as per review comment.
- Remove CamlinternalAtomic. Since atomics are now primitive, declare
  them locally from externals like the rest. This rids us of the wiggle dance of
  dependencies.
- Fix unknown directive warning

No change entry needed.

### CODE REVIEW

File: stdlib/stdlib.ml [hunk 1]
```diff
@@ -547,21 +547,26 @@ let ( ^^ ) (Format (fmt1, str1)) (Format (fmt2, str2)) =
 
 external sys_exit : int -> 'a = "caml_sys_exit"
 
-let exit_function = CamlinternalAtomic.make flush_all
+(* for at_exit *)
+type 'a atomic_t
+external atomic_make : 'a -> 'a atomic_t = "%makemutable"
+external atomic_get : 'a atomic_t -> 'a = "%atomic_load"
+external compare_and_set : 'a atomic_t -> 'a -> 'a -> bool = "%atomic_cas"
```
Comment (gasche) [inline stdlib/stdlib.ml]: `atomic_compare_and_set` or `atomic_cas`?

**Review by gasche — APPROVED** [review]
Useful fix and welcome simplification. Thanks!

**Review by kayceesrk — APPROVED** [review]
Thanks @gadmm. LGTM.

**Comment by jhjourdan** [comment]
Isn't OCaml supposed to be fully Copyright Inria ?

Cc @xavierleroy.

**Comment by xavierleroy** [comment]
> Isn't OCaml supposed to be fully Copyright Inria ?

No.  In particular, we don't ask contributors to transfer copyright to Inria (like FSF does for the GNU projects).  However, we ask contributors to sign a Contributor License Agreement authorizing Inria to redistribute their code under the license of its choice (like the Apache Foundation does for its projects).  See https://github.com/ocaml/ocaml/blob/trunk/CONTRIBUTING.md#contributor-license-agreement .

File: stdlib/atomic.mli [hunk 2]
```diff
@@ -14,17 +15,7 @@
...
-    standard library:
-
-    https://github.com/ocaml-multicore/ocaml-multicore/blob/parallel_minor_gc/stdlib/atomic.mli
-
-    This sequential implementation is provided in the interest of
-    compatibility: when people will start writing code to run on
-    Multicore, it would be nice if their use of Atomic was
-    backward-compatible with older versions of OCaml without having to
-    import additional compatibility layers.
+(** Atomic references
```
Comment (dbuenzli) [inline stdlib/atomic.mli]: This also fixes the module synopsis thanks ! But could you please add a `.` here, all other module synopses have a dot.

File: stdlib/stdlib.ml (hunk 1 above)
Comment (gadmm) [reply stdlib/stdlib.ml]: fixed

File: stdlib/atomic.mli (hunk 2 above)
Comment (gadmm) [reply stdlib/atomic.mli]: fixed, thanks

**Comment by gasche** [comment]
I think that the CI is complaining that you need to `make alldepend`.

**Comment by gadmm** [comment]
Indeed, and indeed it is `alldepend` instead of `depend`. It would have saved me 10 minutes if I had seen your edit.

**Comment by xavierleroy** [comment]
xref: #10861

**Pull request merged**